src/pytest_cov/plugin.py

```diff
@@ -282,16 +284,25 @@ def pytest_terminal_summary(self, terminalreporter):
             terminalreporter.write(message, **markup)
 
     def pytest_runtest_setup(self, item):
+        if self.options.cov_context:
+            context = "{item.nodeid}|setup".format(item=item)
+            self.cov_controller.cov.switch_context(context)
```

Comment:
I don't think plugin should call cov directly. I think this is something that cov_controller should manage

Reply:
eg pass options.cov_context here:

https://github.com/pytest-dev/pytest-cov/blob/master/src/pytest_cov/plugin.py#L157-L165

then run `self.cov_controller.switch_context(item.nodeid, "setup")`

Reply:
Thanks for the feedback.  I should close this PR: it will eventually be folded into #319 (where I could use some help understanding why tests are failing, actually... :) )